wagtail/wagtailimages/tests/test_admin_views.py

```diff
@@ -95,6 +97,21 @@ def test_add_no_file_selected(self):
         # The form should have an error
         self.assertFormError(response, 'form', 'file', "This field is required.")
 
+    @override_settings(WAGTAILIMAGES_MAX_UPLOAD_SIZE=1)
+    def test_add_too_large_file(self):
+        response = self.post({
+            'title': "Test image",
+            'file': SimpleUploadedFile('test.png', get_test_image_file().file.getvalue()),
+        })
+
+        # Shouldn't redirect anywhere
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'wagtailimages/images/add.html')
+
+        # The form should have an error
+        # Note: \xa0 = non-blocking space
+        self.assertFormError(response, 'form', 'file', "This file is too big (1.9\xa0KB). Maximum filesize 1\xa0byte.")
```

Comment:
Hard-coding "1.9 KB" in the test result is a bit of a liability - we can't count on that remaining accurate for every version of PIL / Willow that might be in use.